c/src/ml-api-inference-internal.c

```diff
@@ -297,6 +297,44 @@ __ml_validate_model_file (const char *const *model,
   return ML_ERROR_NONE;
 }
 
+/**
+ * @brief Internal helper function to read the string from specific ini file.
+ */
+static gchar *
+__ml_get_string_from_conf (const gchar * ini_path,
+    const gchar * group, const gchar * key)
+{
+  g_autofree gchar *value = NULL;
+  g_autoptr (GKeyFile) key_file = NULL;
+
+  if (!ini_path || !g_file_test (ini_path, G_FILE_TEST_EXISTS)) {
+    _ml_loge
+        ("The parameter, ini_path, is NULL. It should be a valid path of configuration file.");
+    return NULL;
+  }
+
+  if (!group) {
+    _ml_loge
+        ("The parameter, group, is NULL. It should be a valid string of group name.");
+    return NULL;
+  }
+
+  if (!key) {
+    _ml_loge
+        ("The parameter, key, is NULL. It should be a valid string of key.");
+    return NULL;
+  }
+
+  key_file = g_key_file_new ();
+  g_assert (key_file != NULL);
```

Comment:
Please do not add g_assert in main code; this is a library for an application and applications may want to keep running (with some error handling) with an error in our library.
(you may add assert in test codes.)

Please add a proper error handling code instead.